### PR TITLE
Decouple from the license services that are part of dor-services

### LINKED ISSUE #2601: Decouple from Dor::CreativeCommonsLicenseService

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
fixes #2601
This allows us to decouple from Fedora 3

### CODE REVIEW

File: app/forms/apo_form.rb [hunk 1]
```diff
@@ -100,16 +98,42 @@ def tags_client
...
+                  uri: 'http://opendatacommons.org/licenses/by/1.0/' },
+    'odc-odbl' => { label: 'Open Data Commons Open Database License 1.0',
+                    uri: 'http://opendatacommons.org/licenses/odbl/1.0/' },
+    'by' => { label: 'Attribution 3.0 Unported',
+              uri: 'https://creativecommons.org/licenses/by/3.0/' },
+    'by-sa' => { label: 'Attribution Share Alike 3.0 Unported',
+                 uri: 'https://creativecommons.org/licenses/by-sa/3.0/' },
+    'by_sa' => { label: 'Attribution Share Alike 3.0 Unported',
+                 uri: 'https://creativecommons.org/licenses/by-sa/3.0/',
+                 deprecation_warning: 'license code "by_sa" was a typo in argo, prefer "by-sa"' },
```
Comment (mjgiarlo) [inline app/forms/apo_form.rb:114]: I know you extracted this from dor-services, but I wonder if we can drop this? Could be a follow-up.

File: app/forms/apo_form.rb [hunk 2]
```diff
@@ -100,16 +98,42 @@ def tags_client
...
-        ["#{term.label} (#{term.deprecation_warning})", term.key]
-      elsif term.deprecation_warning.nil?
-        [term.label, term.key]
+  def options_for_use_license_type(current_value)
+    LICENSES.map do |key, attributes|
+      if key == current_value && attributes.key?(:deprecation_warning)
+        ["#{attributes.fetch(:label)} (#{attributes.fetch(:deprecation_warning)})", key]
+      elsif !attributes.key?(:deprecation_warning)
+        [attributes.fetch(:label), key]
       end
```
Comment (mjgiarlo) [inline app/forms/apo_form.rb:137]: If we drop the `by_sa` entry above, we can clean this up. If not, NBD.

**Review by mjgiarlo — APPROVED** [review]
Have a question for you to consider, and pending that, :heavy_check_mark:

File: app/forms/apo_form.rb (hunk 1 above)
Comment (jcoyne) [reply app/forms/apo_form.rb:114]: That seems good to me. I'll drop that change.

Comment (mjgiarlo) [reply app/forms/apo_form.rb:114]: Also checked in QA and stage and _nada_. :beers:

File: app/forms/apo_form.rb (hunk 2 above)
Comment (jcoyne) [reply app/forms/apo_form.rb:137]: Should we keep this code so that we can switch to use URIs and then deprecate all the 3.0 licenses?

Comment (mjgiarlo) [reply app/forms/apo_form.rb:137]: @jcoyne meh. ok w/ whatever.

Comment (mjgiarlo) [reply app/forms/apo_form.rb:137]: @jcoyne still :heavy_check_mark: so I leave that to you!

**Pull request merged**